writer/proto/resource.go

```diff
@@ -150,8 +174,9 @@ func GeneratedResourceMessage(r *parser.ParsedResource, s *parser.ParsedService,
 				LeadingComment: fmt.Sprintf("Field for %v.", p.Name),
 			},
 		)
-		if(p.Type == schema.Type_ARRAY) {
-			f.SetRepeated()
+		switch p.GetTypes().(type) {
```

Comment:
```suggestion
if p.GetTypes().(type) == *schema.Property_ArrayType {
```

Can you do this? slight improvement in readability vs single-case switch.

Reply:
And actually - I think this whole function will be more readable if you just factor the whole construction of the field to it's own method, and make that `protoField` or something along those lines. Then you can actually bundle in the `f.SetRepeated()` call in with the field construction, vs calling it here a bit in a harder to find location. WDYT?

Reply:
I wish we could do this! Types are only accessible in switch statements. It's a strange choice, but here we are!

Reply:
Made the function change. Good call!